echo-pipelinetriggers/echo-pipelinetriggers.gradle

```diff
@@ -27,4 +27,8 @@ dependencies {
   compile spinnaker.dependency("okHttp")
   compile spinnaker.dependency("eurekaClient")
   compile spinnaker.dependency("kork")
+
+  compile group: 'org.codehaus.groovy', name: 'groovy', version: "2.4.5"
```

Comment:
I changed that because I was getting confusing test failures. @tomaslin suggested upgrading Groovy and Spock.

With the following:
```
dependencies {
  spinnaker.group("test")
  spinnaker.group("lombok")
  spinnaker.group("retrofitDefault")

  compile project(':echo-core')
  compile spinnaker.dependency("bootWeb")
  compile spinnaker.dependency("bootActuator")
  compile spinnaker.dependency("rxJava")
  compile spinnaker.dependency("guava")
  compile spinnaker.dependency("okHttp")
  compile spinnaker.dependency("eurekaClient")
  compile spinnaker.dependency("kork")

  compile spinnaker.dependency("groovy")
  spinnaker.group("spockBase")
}
```
my tests fail like so:
```
General error during canonicalization: Comparison method violates its general contract!
java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeLo(TimSort.java:777)
	at java.util.TimSort.mergeAt(TimSort.java:514)
	at java.util.TimSort.mergeCollapse(TimSort.java:441)
	at java.util.TimSort.sort(TimSort.java:245)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1454)
	at java.util.Collections.sort(Collections.java:175)
	at org.codehaus.groovy.transform.trait.TraitComposer.applyTrait(TraitComposer.java:206)
	at org.codehaus.groovy.transform.trait.TraitComposer.doExtendTraits(TraitComposer.java:109)
	at org.codehaus.groovy.control.CompilationUnit$4.call(CompilationUnit.java:188)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1047)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:583)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:561)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:538)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:517)
	at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:164)
	at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:52)
	at org.gradle.api.internal.tasks.compile.daemon.CompilerDaemonServer.execute(CompilerDaemonServer.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```